apps/src/lib/node/ledger/shell.rs

```diff
@@ -495,16 +497,113 @@ impl Shell {
         Default::default()
     }
 
-    /// Check the fees and signatures of the fee payer for a transaction
+    /// Validate a transaction request. On success, the transaction will
+    /// included in the mempool and propagated to peers, otherwise it will be
+    /// rejected.
+    ///
+    /// Checks if the Tx can be deserialized from bytes. Checks the fees and
+    /// signatures of the fee payer for a transaction if it is a wrapper tx.
     ///
     /// INVARIANT: Any changes applied in this method must be reverted if the
     /// proposal is rejected (unless we can simply overwrite them in the
     /// next block).
     pub fn process_proposal(
         &mut self,
-        _req: shim::request::ProcessProposal,
+        req: shim::request::ProcessProposal,
     ) -> shim::response::ProcessProposal {
-        Default::default()
+        let tx = Tx::try_from(req.tx.as_ref()).map_err(Error::TxDecoding);
+        // If we could not deserialize the Tx, return an error response
+        if let Err(err) = tx {
+            return shim::response::ProcessProposal {
+                result: err.into(),
+                tx: req.tx,
+            };
+        }
+
+        let (result, processed_tx) = match process_tx(tx.unwrap()) {
+            // This occurs if the wrapper tx signature is invalid
+            Err(err) => (TxResult::from(err), None),
+            Ok(result) => match result {
+                // If it is a raw transaction, we do no further validation
+                TxType::Raw(_) => (
```

Comment:
this case is temporary until we encrypt all the transactions, right?

Reply:
Yes, in the sense that it is temporary. It allows backwards compatibility at the moment. In the future, we should also have `TxType::Decrypted(_)` and `TxType::UnDecryptable` and `TxType::PVSS` and maybe some others.